README.md

```diff
@@ -37,7 +37,7 @@ The dependencies for this project are in the go.mod file.
 
 To build the source, execute `make clean build`.
 
-To run unit tests, execute `make unit-test`.
+To run unit tests, execute `make unit-test`. Ensure that all dependent libraries and repositories are cloned locally before running unit tests. For example, this includes `gopowermax`, `gocsi`, `gobrick`, `gofsutil`, and others.
```

Comment:
Great! Let the github action pass the check then we will try to have it in place as well as part of this PR only.

Reply:
Yes, that can be done. But this PR mainly enables all the unit tests for better code coverage for the github actions. I have added the code to clone the dependent repos. It is working and I see that the coverage is  ` 75.3 %`.
```
time="2024-10-17T09:48:44Z" level=debug msg="CleanupMapEntries: Current number of entries in lock map: 0"
coverage: 75.3% of statements
status 1
FAIL	github.com/dell/csi-powermax/v2/service	193.782s
FAIL
test failed with return code 1, not proceeding with coverage check

```

Some tests are failing that need to be fixed.

Reply:
Putting few suggestions here; in case someone look into this:

<img width="871" alt="image" src="https://github.com/user-attachments/assets/d296b167-883c-452a-a553-45742c5b9bda">

We can clone the repo beforehand and proceed, considering if repo was cloned earlier, it will report the error but test will continue with the already cloned repo.

<img width="952" alt="image" src="https://github.com/user-attachments/assets/5a69c453-0d9f-42b6-a202-04f11205725b">


<img width="952" alt="image" src="https://github.com/user-attachments/assets/13454bbf-aff1-4923-a1de-56e1cccf09e2">

Reply:
Need to assess, how this actually affect the github actions as we have already cloned it as part of action's steps.